tests/tests/regenesis.rs

```diff
@@ -0,0 +1,211 @@
+use std::net::Ipv4Addr;
+
+use clap::Parser;
+use fuel_core_bin::cli::{
+    run,
+    snapshot,
+};
+use fuel_core_client::client::{
+    pagination::{
+        PageDirection,
+        PaginationRequest,
+    },
+    FuelClient,
+};
+use fuel_core_types::{
+    fuel_tx::*,
+    fuel_vm::{
+        checked_transaction::IntoChecked,
+        *,
+    },
+};
+use rand::SeedableRng;
+use tempfile::{
+    tempdir,
+    TempDir,
+};
+use tokio::task::{
+    self,
+    JoinHandle,
+};
+
+pub struct FuelCoreDriver {
+    pub port: u16,
+    /// This must be before the db_dir as the drop order matters here
+    pub process: JoinHandle<anyhow::Result<()>>,
+    pub db_dir: TempDir,
+    pub client: FuelClient,
+}
+impl FuelCoreDriver {
+    pub async fn spawn(extra_args: &[&str]) -> Self {
+        // Generate temp params
+        let db_dir = tempdir().expect("Failed to create temp dir");
+        let port = portpicker::pick_unused_port().expect("Failed to pick unused port");
+
+        let port_str = port.to_string();
+        let mut args = vec![
+            "_IGNORED_",
+            "--port",
+            &port_str,
+            "--db-path",
+            db_dir.path().to_str().unwrap(),
+        ];
+        args.extend(extra_args);
+
+        let process = task::spawn(run::exec(run::Command::parse_from(args)));
```

Comment:
Hmm, do you really need to use `run::Command` to start a node? It seems you still can use `FuelService::new_node` to do that and work directly with the `SharedState.graph_ql.bound_address` to send transactions and get data. In this case you don't need to specify the port. 

Or you can split the `run::exec`command into two functions where one function has the logic around stop signal:

<img width="870" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/653534cc-b169-4173-bed3-ffc4af176b3c">


And another one contains all initialisation logic:
<img width="911" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/06b953bc-6e1b-4428-95b9-d0a3b18eb2d1">

Reply:
Yep, done in b630cd9a3